src/Microsoft.DotNet.Build.CloudTestTasks/AzureHelper.cs

```diff
@@ -0,0 +1,373 @@
+﻿// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="AzureHelper.cs" company="">
+//   
+// </copyright>
+// <summary>
+//   The azure helper.
+// </summary>
+// --------------------------------------------------------------------------------------------------------------------
+
+namespace Microsoft.DotNet.Build.CloudTestTasks
+{
+    #region
+
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Net;
+    using System.Security.Cryptography;
+    using System.Text;
+    using System.Text.RegularExpressions;
+
+    #endregion
+
+    /// <summary>
+    /// The azure helper.
+    /// </summary>
+    public static class AzureHelper
+    {
+        /// <summary>
+        /// The storage api version.
+        /// </summary>
+        public static readonly string StorageApiVersion = "2015-04-05";
+
+        /// <summary>
+        /// The sas access type.
+        /// </summary>
+        public enum SasAccessType
+        {
+            /// <summary>
+            /// The read.
+            /// </summary>
+            read, 
+
+            /// <summary>
+            /// The write.
+            /// </summary>
+            write
+        };
+
+        /// <summary>
+        /// The authorization header.
+        /// </summary>
+        /// <param name="storageAccount">
+        /// The storage account.
+        /// </param>
+        /// <param name="storageKey">
+        /// The storage key.
+        /// </param>
+        /// <param name="method">
+        /// The method.
+        /// </param>
+        /// <param name="now">
+        /// The now.
+        /// </param>
+        /// <param name="request">
+        /// The request.
+        /// </param>
+        /// <param name="ifMatch">
+        /// The if match.
+        /// </param>
+        /// <param name="md5">
+        /// The md 5.
```

Comment:
"The MD5 hash of [whatever it's hashing], as [I'm assuming base64, but maybe hex?]"?